src/algorithms/floyd-steinberg.cpp

```diff
@@ -33,29 +33,43 @@ QImage* FloydSteinberg::createMosaic(const QImage& input, int maxNumOfMolluscs)
 
     auto result = new QImage(width * molluscSize, height * molluscSize, input.format());
     result->fill(Qt::GlobalColor::white);
-    QPainter painter(result);
+    QPainter resultPainter(result);
+    auto idTexture = new QImage(width * molluscSize, height * molluscSize, input.format());
+    idTexture->fill(Qt::GlobalColor::black);
+    QPainter idPainter(idTexture);
 
+    auto i = 0;
     for (auto y = 0; y < scaled.height(); ++y)
     {
         for (auto x = 0; x < scaled.width(); ++x)
         {
             auto oldColor = QColor(scaled.pixel(x, y));
             auto oldVector = toVec3(oldColor) + errorStorage[x + 1 + y * width];
 
-            const Mollusc& mollusc = m_molluscPalette.getClosestColor(oldVector);
-            auto newVector = toVec3(mollusc.m_color);
+            auto mollusc = m_molluscPalette.getClosestColor(oldVector);
+            auto newVector = toVec3(mollusc->m_color);
 
-            if (mollusc.m_imageName.compare("NONE") != 0)
-                painter.drawPixmap(x * molluscSize, y * molluscSize, molluscSize, molluscSize, mollusc.m_image);
+            if (mollusc->m_imageName.compare("NONE") != 0) {
```

Comment:
is this supposed to be the same code as the painter? if you don't want to update floyd-steinberg to use the new interface now, just create an issue so we don't forget it.

Reply:
floyd-steinberg is updated now